mobile-app/app/screens/AppNavigator/screens/Balances/hooks/TokenPrice.ts

```diff
@@ -50,19 +58,24 @@ export function useTokenPrice (): DexTokenPrice {
       const usdTokenB = getTokenPrice(pair.data.tokenB.symbol, tokenBAmount.toFixed(8))
       return usdTokenA.plus(usdTokenB)
     }
+    const key = `WALLET.${network}.${blockCount ?? 0}.TOKEN_PRICE_${symbol}`
+    const result = CacheApi.get(key)
+    if (result !== undefined) {
+      return new BigNumber(result).multipliedBy(amount)
+    }
     // active price for walletTokens based on USDT
     const arbitraryPoolPair = getArbitraryPoolPair(symbol, 'USDT')
 
-    if (symbol === 'USDT') {
-      return new BigNumber(amount)
-    } else if (arbitraryPoolPair.length > 0) {
-      const { estimated } = calculatePriceRates(symbol, arbitraryPoolPair, amount)
+    if (arbitraryPoolPair.length > 0) {
+      const { aToBPrice, estimated } = calculatePriceRates(symbol, arbitraryPoolPair, amount)
+      // store price for each unit in cache
+      CacheApi.set(key, aToBPrice.toFixed(8))
```

Comment:
Just want to clarify if the intent here it to keep cache for 10mins since that is the default `maxAge`? If the tokenReserve change drastically before that, then the price rates will use the cached one. Thus, we will be displaying inaccurate rates for 10mins

Reply:
Oops. sorry I missed the key has the blockCount in it

Reply:
Maybe we can reduce the default maxAge since a new block is minted every 30seconds so there's no reason to keep the `expired` ones in the device's memory for 10mins? Do you think 2 minutes is more reasonable as `maxAge`

Reply:
@pierregee sometimes our blockchain takes 5-6 min to block a new block(based on historic data), But to lower memory usage we will set it to 5